oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/relations/OSHDBGeometryBuilderMultipolygonInvalidInnersTest.java

```diff
@@ -25,10 +25,18 @@ public OSHDBGeometryBuilderMultipolygonInvalidInnersTest() {
   }
 
   @Test
-  public void test() {
+  public void testDuplicateInnerRings() {
     // data has invalid (duplicate) inner rings
     OSMEntity entity = testData.relations().get(1L).get(0);
     Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, tagInterpreter);
     assertTrue(result instanceof Polygon);
   }
+
+  @Test
+  public void testTouchingIncompleteInnerRings() {
+    // data has invalid (duplicate) inner rings
+    OSMEntity entity = testData.relations().get(2L).get(0);
+    Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, tagInterpreter);
+    assertTrue(result instanceof Polygon);
```

Comment:
Same question here. What would we expect with an invalid second inner “ring” and a valid ring? Or is this already covered by another test?

Reply:
see https://github.com/GIScience/oshdb/pull/343#discussion_r565943183

IMHO, we shouldn't care. There would be so many possible ways to handle invalid data, in principle. Sure, some perhaps "better" ways and some worse ways, but if we restrict us too much here we loose quite some flexibility in the implementation. And I can live with some ambiguity in the output if the input is not really valid in the first place.